File: programovanie/builder/prikl/graf/unit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#include <math.h>
#pragma hdrstop

#include "Unit1.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
int pu;
double xx,yy;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TForm1::Button1Click(TObject *Sender)
{  pu=StrToInt(Edit1->Text);
   StringGrid1->ColCount = pu+1;
   StringGrid1->RowCount = pu+1;
   for (int i=1; i<=pu; i++)
   {  StringGrid1->Cells[0][i]=i;
      StringGrid1->Cells[i][0]=i;
      xx=300 + 200*cos(2*3.14/pu*i);
      yy=300 + 200*sin(2*3.14/pu*i);
      Image1->Canvas->Ellipse(xx-10,yy-10,xx+10,yy+10);
   }
   for (int i=1; i<=pu; i++)
     for (int j=1;j<=pu;j++)
        StringGrid1->Cells[i][j] = 0;


}
//---------------------------------------------------------------------------
 

File: programovanie/builder/06tabulka/ulohy/u1/unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, Grids;

type
  TForm1 = class(TForm)
    StringGrid1: TStringGrid;
    Button1: TButton;
    procedure Button1Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure aaa(Sender: TObject; ACol, ARow: Integer; Rect: TRect;
      State: TGridDrawState);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

procedure TForm1.Button1Click(Sender: TObject);
begin
StringGrid1.Cells[2,1]:='ahoj';
StringGrid1.Canvas.TextRect(StringGrid1.Cellrect(2,1),1,20,'ahoj');
end;

procedure TForm1.FormCreate(Sender: TObject);
begin
//
end;

procedure TForm1.aaa(Sender: TObject; ACol, ARow: Integer; Rect: TRect;  State: TGridDrawState);
begin
StringGrid1.Canvas.Ellipse(StringGrid1.CellRect(Acol,Arow));
//StringGrid1.Canvas.Ellipse(Rect);
StringGrid1.Canvas.TextRect(Rect, Rect.Left+20, Rect.Top+7, StringGrid1.Cells[ACol, ARow]);
StringGrid1.Canvas.Brush.Style:=bsClear;
StringGrid1.Canvas.Ellipse(StringGrid1.CellRect(Acol,Arow));
end;

end.
